src/platform/renesas/DiagnosticDataProviderImpl.cpp

```diff
@@ -93,6 +94,118 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(BootReasonType & bootReason
     return err;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface ** netifpp)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    NetworkInterface *head = nullptr;
+
+    for (Inet::InterfaceIterator interfaceIterator; interfaceIterator.HasCurrent(); interfaceIterator.Next())
+    {
+        NetworkInterface *ifp = new NetworkInterface();
+        interfaceIterator.GetInterfaceName(ifp->Name, Inet::InterfaceId::kMaxIfNameLength);
+        ifp->name          = CharSpan::fromCharString(ifp->Name);
+        if (ifp == nullptr)
```

Comment:
This should be moved up to before the name now or it will crash before checking for memory errors

Reply:
updated

Reply:
not far enough. in line 105 `ifp` is dereferenced :boom:

Reply:
updated